tendermint/src/rpc/event_listener.rs

```diff
@@ -73,9 +73,23 @@ impl EventListener {
             .await
             .ok_or_else(|| RPCError::websocket_error("web socket closed"))??;
         match serde_json::from_str::<JsonRPCBlockResult>(&msg.to_string()) {
-            Ok(data) => Ok(Event::JsonRPCBlockResult(data)),
+            Ok(data) => {
+                if let Some(data) = data.0.result {
+                    return Ok(Event::JsonRPCBlockResult(data));
+                } else {
+                    // The Websocket should never send an empty block
+                    panic!("Websocket sent us an empty block")
```

Comment:
I think we can leave this as is but I'm a bit worried that someone could send a non-valid response and crash the process (even if tendermint wouldn't send an empty block here). As far as I see the connection isn't authenticated so this should be fairly easy.

Reply:
b11ea9a